python/pyspark/sql/connect/client.py

```diff
@@ -145,6 +145,39 @@ def _build_metrics(self, metrics: "pb2.Response.Metrics") -> typing.List[PlanMet
     def sql(self, sql_string: str) -> "DataFrame":
         return DataFrame.withPlan(SQL(sql_string), self)
 
+    def range(
+        self,
+        start: int,
+        end: int,
+        step: Optional[int] = None,
```

Comment:
i think we can use `step: int = 1`

Reply:
furthermore, i think we can make `step` a required field in proto

Reply:
hmmm we are not marking `step` as required because Scala side implementation does not treat it as required and it has a default value.

```
  private def transformRange(rel: proto.Range): LogicalPlan = {
    val start = rel.getStart
    val end = rel.getEnd
    val step = if (rel.hasStep) {
      rel.getStep.getStep
    } else {
      1
    }
    val numPartitions = if (rel.hasNumPartitions) {
      rel.getNumPartitions.getNumPartitions
    } else {
      session.leafNodeDefaultParallelism
    }
    logical.Range(start, end, step, numPartitions)
  }
```